### ISSUE TITLE
Support cabal only projects

### ISSUE THREAD

**Issue by typetetris** [rikvdkleij/intellij-haskell#625]
It would be great, if cabal only projects would be supported too.

Sometimes its cumbersome  to get a project to work in one of the package sets stack offers and
you sometimes want to have the most recent versions possible of your dependencies.

Maybe just sketch what needs to be done, so maybe someone else could try it?

**Comment by rikvdkleij** [comment]
Yeah, that is already for a long time on my wish list. Some time ago I already reduced the stack dependencies.

Have to investigate how much code is affected. I guess the challenge is to keep the codebase maintainable for supporting Cabal and Stack projects. So to be continued 😄

**Comment by typetetris** [comment]
Hey there!
How is it going?

**Comment by rikvdkleij** [comment]
@typetetris no progress....

**Comment by coderfromhere** [comment]
VS Code is now supporting the language server through [this plugin](https://github.com/haskell/vscode-haskell), which in turn  supports pure Cabal projects. If not the Idea diff/merge tooling, I'd say that it now should be the default choice for anyone looking to get an IDE experience with pure Cabal projects. But I still hope that Idea gets a proper Cabal support too, especially since Cabal V2 API - Stack has very little to offer in addition to it, and for those who prefer yaml-style configurtions, there's `hpack` that can be used independently from Stack.

**Comment by declension** [comment]
Any ideas / pointers on how this might get done? 

We've got some projects that are Nix-based (i.e. Cabal) and we really want to use intellij-haskell to work on them :+1:
